File: Problems/692_Top_K_Frequent_Words.py
class Solution:
    def topKFrequent(self, words: List[str], k: int) -> List[str]:
        hashMap = {}
        hp = []
        for word in words:
            if word in hashMap:
                hashMap[word] += 1
            else:
                hashMap[word] = 0
        
        for word, count in hashMap.items():
            heapq.heappush(hp, (-count, word))
        res = []
        for _ in range(k):
            res.append(heapq.heappop(hp)[1])
        return res

File: Problems/692_Top_K_Frequent_Words.cpp
class Solution {
public:
    vector<string> topKFrequent(vector<string>& words, int k) {
        map<string,int> hashMap;
        priority_queue<pair<int, string>> pq;
        for(auto word: words) hashMap[word]++;
        for(auto el: hashMap){
            pq.push({-el.second, el.first});
            if(pq.size()>k) pq.pop();
        }
        vector<string> ans;
        while(!pq.empty()){
            ans.push_back(pq.top().second);
            pq.pop();
        }
        
        reverse(ans.begin(),ans.end());
        return ans;
    }
};